java-frontend/src/main/java/org/sonar/java/JavaClasspath.java

```diff
@@ -52,6 +52,9 @@ protected void init() {
         throw new AnalysisException(
           "sonar.binaries and sonar.libraries are not supported since version 4.0 of sonar-java-plugin, please use sonar.java.binaries and sonar.java.libraries instead");
       }
+      if (binaries.isEmpty() && hasMoreThanOneJavaFile()) {
+        throw new AnalysisException("Please provide compiled classes of your project with sonar.java.binaries property");
```

Comment:
I would put quote around `sonar.java.binaries` to highlight it and make it explicit in the middle of the error message.